core/platform/models_test.py

```diff
@@ -0,0 +1,271 @@
+        expected_user_models = (user_models,)
+        self.assertEqual(
+            expected_user_models,
+            self.registry_instance.import_models([models.NAMES.user]))
+
+    def test_import_models_invalid(self):
+        """Tests import_models function with an invalid option."""
+        with self.assertRaises(Exception):
+            self.registry_instance.import_models([''])
+
+    def test_import_current_user_services(self):
+        """Tests import current user services function."""
+        from core.platform.users import gae_current_user_services
+        self.assertEqual(
+            self.registry_instance.import_current_user_services(),
+            gae_current_user_services)
+
+    def test_import_datastore_services(self):
+        """Tests import datastore services function."""
+        from core.platform.datastore import gae_datastore_services
+        self.assertEqual(
+            self.registry_instance.import_datastore_services(),
+            gae_datastore_services)
+
+    def test_import_transaction_services(self):
+        """Tests import transaction services function."""
+        from core.platform.transactions import gae_transaction_services
+        self.assertEqual(
+            self.registry_instance.import_transaction_services(),
+            gae_transaction_services)
+
+    def test_import_app_identity_services(self):
+        """Tests import app identity services function."""
+        from core.platform.app_identity import gae_app_identity_services
+        self.assertEqual(
+            self.registry_instance.import_app_identity_services(),
+            gae_app_identity_services)
+
+    def test_import_gae_image_services(self):
+        """Tests import gae image services function."""
+        from core.platform.image import gae_image_services
+        self.assertEqual(
+            self.registry_instance.import_gae_image_services(),
+            gae_image_services)
+
+    def test_import_email_services_gae(self):
+        """Tests import email services method for when email service provider is
+        the default - GAE.
+        """
+        from core.platform.email import gae_email_services
+        self.assertEqual(
+            gae_email_services, self.registry_instance.import_email_services())
+
+    def test_import_email_services_mailgun(self):
+        """Tests import email services method for when email service provider is
+        mailgun.
+        """
+        feconf.EMAIL_SERVICE_PROVIDER = feconf.EMAIL_SERVICE_PROVIDER_MAILGUN
+        from core.platform.email import mailgun_email_services
+        self.assertEqual(
+            mailgun_email_services,
+            self.registry_instance.import_email_services())
+
+    def test_import_email_services_invalid(self):
+        """Tests import email services method for when email service provider is
+        an invalid option.
+        """
+        feconf.EMAIL_SERVICE_PROVIDER = 'invalid service provider'
```

Comment:
Use "with self.swap" here -- don't set directly. See other examples where "with self.swap" is used in the codebase.

Reply:
Done!